vonk/deployment/performance.rst

```diff
@@ -0,0 +1,161 @@
+#. SQL Server: Performance tuning of SQL Server is a topic on its own. Vonk manages the tables it needs, and the indexes on top of it are optimized for the way Vonk queries them.
+#. MongoDB: Performance tuning of MongoDB is a topic on its own. Vonk manages the collections it needs, and the indexes on top of it are optimized for the way Vonk queries them. MongoDB is used in our own performance test, see below.
+
+Prevalidation
+~~~~~~~~~~~~~
+
+By default, a resource that is sent to Vonk is fully validated against its StructureDefinition. This requires extra processing and thus extra time. But you can disable full validation if needed, with the :ref:`ValidateIncomingResources <feature_prevalidation>` setting. We have no test yet in place to time the difference caused by this setting.
+
+Search Parameters
+~~~~~~~~~~~~~~~~~
+
+When a resource is sent to Vonk for storage, Vonk indexes the resource for all the search parameters that are applicable to the resource. The more search parameters are known to Vonk, the more resources will be used for indexing. Both time for the indexing processing and storage for storing the values for each search parameter. This also increases the size of the index tables and indexes, and therefore querying times. So if you know you will use only part of the predefined search parameters you can choose to delete the others from the Administration API. See :ref:`conformance` and :ref:`supportedmodel`.
+
+Pipeline
+~~~~~~~~
+
+Vonk is made up of a pipeline of components. You can leave out any component that you don't need. So if you don't need conditional processing (create, update, delete), just exclude them from the pipeline. Excluded components are not loaded and thus never executed. See :ref:`settings_pipeline`.
+
+Platform
+^^^^^^^^
+
+Vonk can run on Windows, Linux and MacOS. Directly or in a Docker container. On real hardware, virtual machine, app service or Kubernetes cluster. And then you can choose the dimensions of the platform, scaling up (bigger 'hard'ware) or scaling out (more (virtual) machines) as you see fit. Each of these choices will influence performance.
+
+Besides the way Vonk is deployed, the way the database is deployed is an important factor. Vonk needs a very low latency connection between the Vonk process(es) and the database server. If you have configured :ref:`configure_log_insights`, the calls to the database are recorded as separate dependencies so you can check whether this may be a bottleneck.
+
+Vonk is optimized for multithreaded processing. This means that it will benefit from extra processing cores, either in the same machine (multi core processor) or by adding additional machines (and thus processors). 
+
+Vonk is fully stateless, so no data about connections is saved between requests. First of all this helps in scaling out, since you don't need thread affinity. On top of that this limits the memory requirements of Vonk.
```

Comment:
```suggestion
Vonk is fully stateless, so no data about connections is saved between requests. First of all this helps in scaling out, since you don't need thread affinity. On top of that this reduces the memory requirements of Vonk.
```